franmolmedo/react-context-and-portals/readme.md

```diff
@@ -0,0 +1,298 @@
+![Header](images/react.png)
+
+# React: Context API y Portals
+
+Cuando abrazas una tecnología o forma de hacer las cosas es agradable ver como cada cierto tiempo van introduciendo sucesivas mejoras que hacen que ciertas limitaciones que has ido encontrando dejen de serlo.
+La comunidad en torno a React es impresionante. No sólo los creadores originales de la librería (Facebook), sino infinidad de personas que han aportado su granito de arena para hacer que la experencia de desarrollo y, sobre todo, los resultados sean muy buenos. Librerías como [Flux](https://github.com/facebook/flux), [Redux](https://github.com/reactjs/redux), [Apollo](https://github.com/apollographql/react-apollo), [Formik](https://github.com/jaredpalmer/formik), [redux-saga](https://github.com/redux-saga/redux-saga)... nos hacen la vida más sencilla a aquellos que trabajamos dia a día con todas ellas.
+Para este mismo año se han anunciado nuevas mejoras (como el [async rendering](https://www.youtube.com/watch?v=v6iR3Zk4oDY&t=230s)) muy interesantes para aplicar en nuestras aplicaciones.
+En esta pequeña entrada me gustaría hablar sobre dos de los últimos añadidos a React, que son el context API y los React Portals. Empecemos.
+
+## Preparando el entorno
+
+Para hacer las pruebas vamos a utilizar la última versión disponible de React, en este momento (16.03.alpha1). Para ello, partimos de [create-react-app](https://github.com/facebook/create-react-app), el starter amparado por Facebook para aplicaciones React que nos monta la estructura básica de la aplicación. Nos vamos a nuestra consola y creamos la aplicación de ejemplo:
+
+```javascript
+npx create-react-app react-context-portals-example
+```
+
+![create-react-app](images/create-react-app.png)
+
+Nos situamos en la carpeta que nos ha creado y abrimos la aplicación en nuestro editor favorito. Create React App nos crea una aplicación React bastante estándar.
+![app-structure](images/app-structure.png).
+
+Vamos a abrir el fichero package.json y vamos a actualizar la versión de los paquetes correspondientes a react y a react-dom. En concreto, vamos a utilizar, como ya hemos comentado, la última disponible en estos momentos. El fichero tiene que quedar de la siguiente forma:
+
+```javascript
+{
+    "name": "react-context-portals-example",
+    "version": "0.1.0",
+    "private": true,
+    "dependencies": {
+        "react": "^v16.3.0-alpha.3",
+        "react-dom": "^v16.3.0-alpha.3",
+        "react-scripts": "1.1.1"
+    },
+    "scripts": {
+        "start": "react-scripts start",
+        "build": "react-scripts build",
+        "test": "react-scripts test --env=jsdom",
+        "eject": "react-scripts eject"
+    }
+}
+```
+
+Una vez hecho esto, nos vamos a la consola y ejecutamos **_yarn start_** para que se nos lance la aplicación en nuestro navegador:
```

Comment:
Change \**\__yarn start__** for \`yarn start\`

Reply:
It is in italic